src/validators/FileUploadValidator.ts

```diff
@@ -0,0 +1,65 @@
+import fs from 'fs';
+import { parse } from 'csv-parse';
+import { FileUpload } from '../models/FileUpload';
+import { Writable } from 'stream';
+
+export const validateFileUpload = (file: string): Promise<string[]> => {
+    return new Promise((resolve, reject) => {
+        const csv = file;
+        const headers = ["roleName", "description", "responsibilities", "linkToJobSpec", "capability", "band", "closingDate", "status", "positionsAvailable", "locations"];
+
+        const fileStream = fs.createReadStream(file);
+
+        const parser = parse({
+            delimiter: ',',
+            columns: headers,
+            skip_empty_lines: true
+        });
+
+        const validationErrors: string[] = [];
+        let rowCount = 0;
+
+        const rowProcessor = new Writable({
+            objectMode: true,
+            write(row, encoding, callback) {
+                rowCount++;
+                const rowErrors: string[] = [];
+
+                if (!row.roleName) rowErrors.push(`Row ${rowCount}: 'roleName' is required`);
+                if (!row.description) rowErrors.push(`Row ${rowCount}: 'description' is required`);
+                if (!row.responsibilities) rowErrors.push(`Row ${rowCount}: 'responsibilities' is required`);
+                if (!row.linkToJobSpec) rowErrors.push(`Row ${rowCount}: 'link to job role' is required`);
+                if (!row.capability) rowErrors.push(`Row ${rowCount}: 'capability' is required`);
+                if (!row.band) rowErrors.push(`Row ${rowCount}: 'band' is required`);
+                if (row.band && isNaN(parseInt(row.band))) {
+                    rowErrors.push(`Row ${rowCount}: 'band' must be a number`);
+                }
+                // if (row.positionsAvailable && isNaN(parseInt(row.positionsAvailable))) {
+                //     rowErrors.push(`Row ${rowCount}: 'positionsAvailable' must be a number`);
+                // }
+                if (!row.closingDate) rowErrors.push(`Row ${rowCount}: 'closing date' is required`);
+
```

Comment:
You are missing validation for locations, positions available and status

Reply:
Yup I realised - told myself that I will do it later :D